### DISCUSSION TITLE
Adding Copyright and License Nodes

### DISCUSSION THREAD

**Discussion by tajmone** [pml-lang/pml-companion#29] [Ideas]
Documents have copyright, and often an open source license too.

I would be really great if PML would allow specifying the Copyright status and license of a document, via some metadata nodes which the default template could expose both internally, in the generated HTML metadata section, and in the document itself, e.g. via some nice badge icon, etc.

I'm actually surprised that the most famous markup syntaxes don't actually support this, along with authors, date, and other metadata.

# Foss Licenses Open Standards

I wonder if PML could make use of open standard like [SPDX]  (Software Package Data Exchange) to handle license info.

# License Badges

For nice examples of license badges, here are some links to various projects and online badge generation services:

- [Shields] — various badge generator libraries, and the online service:
    + [Shields.io] » [Licenses]
- [Markdown License badges] — a Gist with CC or [Shields.io] badges links to the most common open source and open data licenses.

[SPDX]: https://spdx.dev/resources/use/

[Markdown License badges]: https://gist.github.com/lukas-h/2a5d00690736b4c3a7ba

[Shields]: https://github.com/badges "Shields GitHub organization"
[Shields.io]: https://shields.io
[Licenses]: https://shields.io/category/license

**Comment by pml-lang** [comment]
> allow specifying the Copyright status and license of a document, via some metadata nodes

Excellent idea (again)! Besides `copyright` and `license`, we should indeed add more metadata (e.g. `tags`, `keywords`, `abstract`, etc.). And allow users to add their own (with optional icon paths), all nicely displayed at the start of the document.

> some nice badge icon

People would love to see a nice badge, automagically displayed in their documents.

Thanks for the links. I'll look into it.
